lib/lab_manager/workers/action_worker.rb

```diff
@@ -182,6 +183,26 @@ def take_snapshot_vm
       action.save!
     end
 
+    def revert_snapshot_vm
+      lock { compute.revert! }
+      begin
+        fail 'Wrong action payload' unless action.payload
+        fail 'Wrong action payload, no snapshot_id provided' unless action.payload.has_key?(:snapshot_id)
+        snapshot = compute.snapshots.find(action.payload[:snapshot_id])
+        # we suppose, that revert process should be fully finished before performing
+        # some actions such as restart, shutdown, reboot, power_off etc.
+        lock do
+          compute.revert_snapshot_vm( { name: snapshot.name } )
```

Comment:
what happens when we have got snapshot form stopped machine and revert to it when machine is running?
Shouldn't we have the current state and set the compute model state accordingly?